core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.java

```diff
@@ -112,16 +115,11 @@ protected void onCreate(Bundle savedInstanceState) {
       String body = getBody();
 
       if (allowLogsCheckbox.isChecked()) {
-        File appDirectory = new File(Environment.getExternalStorageDirectory() + "/Kiwix");
-        File logFile = new File(appDirectory, "logcat.txt");
-        if (logFile.exists()) {
-          Uri path =
-            FileProvider.getUriForFile(this,
-              getApplicationContext().getPackageName() + ".fileprovider",
-              logFile);
-          emailIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
-          emailIntent.putExtra(Intent.EXTRA_STREAM, path);
-        }
+        File file = fileLogger.writeLogFile(this);
+        Uri path = FileProvider.getUriForFile(this,
```

Comment:
* As per the latest commit, the file is stored in the app's private storage, this has been verified.
* But when an intent is passed for log report from the help section for diagnostic report, the app becomes unresponsive for a while and when fired with multiple same events, it crashes saying
*  `java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority org.kiwix.kiwixmobile.fileProvider`

* I've verified the permissions of fileProvider from manifest and didn't find anything suspicious
* Any ideas on this?

Reply:
I would say you need to update the paths of the FileProvider like so https://developer.android.com/reference/kotlin/androidx/core/content/FileProvider#specifying-available-files